include/xtensor/xstridedview.hpp

```diff
@@ -0,0 +1,722 @@
+
+        using iterable_base = xexpression_iterable<self_type>;
+        using inner_shape_type = typename iterable_base::inner_shape_type;
+        using shape_type = inner_shape_type;
+        using strides_type = shape_type;
+        using closure_type = const self_type;
+
+        using stepper = typename iterable_base::stepper;
+        using const_stepper = typename iterable_base::const_stepper;
+
+        using broadcast_iterator = typename iterable_base::broadcast_iterator;
+        using const_broadcast_iterator = typename iterable_base::const_broadcast_iterator;
+
+        using iterator = typename iterable_base::iterator;
+        using const_iterator = typename iterable_base::const_iterator;
+
+        using temporary_type = typename xcontainer_inner_types<self_type>::temporary_type;
+        using base_index_type = xindex_type_t<shape_type>;
+
+        template <class I>
+        xstrided_view(CT e, I&& shape, I&& strides, std::size_t offset) noexcept;
+
+        template <class E>
+        self_type& operator=(const xexpression<E>& e);
+
+        template <class E>
+        disable_xexpression<E, self_type>& operator=(const E& e);
+
+        size_type size() const noexcept;
+        size_type dimension() const noexcept;
+        const shape_type& shape() const noexcept;
+        const strides_type& strides() const noexcept;
+
+        reference operator()();
+        template <class... Args>
+        reference operator()(Args... args);
+        reference operator[](const xindex& index);
+        reference operator[](size_type i);
+
+        template <class It>
+        reference element(It first, It last);
+
+        const_reference operator()() const;
+        template <class... Args>
+        const_reference operator()(Args... args) const;
+        const_reference operator[](const xindex& index) const;
+        const_reference operator[](size_type i) const;
+
+        template <class It>
+        const_reference element(It first, It last) const;
+
+        template <class O>
+        bool broadcast_shape(O& shape) const;
+
+        template <class O>
+        bool is_trivial_broadcast(const O& /*strides*/) const noexcept;
+
+        template <class ST>
+        stepper stepper_begin(const ST& shape);
+        template <class ST>
+        stepper stepper_end(const ST& shape);
+
+        template <class ST>
+        const_stepper stepper_begin(const ST& shape) const;
+        template <class ST>
+        const_stepper stepper_end(const ST& shape) const;
+
+        underlying_container_type& data() noexcept;
+        const underlying_container_type& data() const noexcept;
+
+        value_type* raw_data() noexcept;
+        const value_type* raw_data() const noexcept;
+
+        size_type raw_data_offset() const noexcept;
+
+    private:
+
+        CT m_e;
+        const shape_type m_shape;
+        const strides_type m_strides;
+        const std::size_t m_offset;
```

Comment:
Why making these members `const` ? This prevents you from reshaping and transposing in place. Not sure we want to allow that, though.